### ISSUE TITLE
Pub dry run meta package issue on Flutter

### ISSUE THREAD

**Issue by anders-sandholm** [dart-lang/pub#1830]
_From @lukef on March 5, 2018 17:49_

```
Dart VM version: 2.0.0-dev.30.0 (Thu Feb 22 14:08:47 2018 +0100) on "macos_x64"
Flutter: master
```

If you use the `@required` annotation and combine it with `import 'package:meta/meta.dart';` then `pub publish --dry-run` will report that the `meta` package is not used. If it is replaced with `import 'package:flutter/foundation.dart';` then the problem goes away.

_Copied from original issue: dart-lang/sdk#32418_

**Comment by matanlurey** [comment]
@lukef: Do you have `package:meta` in your `dependencies: ...` (`pubspec.yaml`)?

Even though Flutter provides some of them in `foundation.dart`, it's considered a warning to _not_ depend directly on `package:meta`. Was that intentional?

**Comment by lukef** [comment]
@matanlurey - I don't have that package in my `pubspec`. That's good to know. I was using `foundation.dart` was for the `@required` annotation so `foundation` is probably fine in this case.

This is more a Flutter issue (which I raised over there) but, in general, its one of those cases where it's semi-unclear which one to use because dart + flutter overlap. My mindset was to use the dart package but, in hindsight, I don't have a super good reason for why I picked that one. Is there guidance either way? I'd be happy to submit a Flutter documentation PR / Issue also.
